### ISSUE TITLE
Stygarctus release plans

### ISSUE THREAD

**Issue by cwgoes** [anoma/juvix#890]
Summary of work below. For more detailed issue tracking, see [the milestone](https://github.com/anoma/juvix/milestone/3).

- Datatypes & pattern matching (@rokopt, @andy-morris, @jonaprieto, @Acentelles) (see #910):
  - Unifier
     - #768  
  - Typecheck entire function defs --> typecheck whole program
    - mostly typechecking patterns & extending the context for the rhs
  - Erasure support
    - Take out zero fields
    - Take out zero-usage fields in pattern matching
  - Backend support
    - Compile datatypes to in-memory representation, pattern matching to operations
    - Compile function clauses to case trees
  - Termination (and coverage) checking
    - Termination checking should happen after typechecking but before erasure
  - Working records (Jeremy, Andy, Jonathan, Alberto)
    - Need a solution without requiring typechecking
    - For now, treat unknown cases as potential record lookups, resolve later
    - #893 
    - ~~#892~~ #927
    - #895
- Basic unification (@andy-morris, @cyberglot)
  - First-order unification
- Basic LLVM backend (@systemf-frank, @mariari, @cwgoes)
  - Lambda calculus (blocked on the bugs below)
- Basic Witch integration (@cyberglot, @mariari, @andy-morris)
  - Syntax  #722 
  - Pipeline transformations #989
  - ANF (old, closed PR #787, new PR to come)
  - Simple tactics / Standard library  #888 
  - Unification / typechecker API #903 
- Misc fixes
  - Nested operations (@Acentelles) 
    - #900
    - #889
  - Move stdlib to standard location (can be overridden), avoid hardcoded path
  - #907 / #909 (@systemf-frank)

**Bugs**
- #806 / #805 (@systemf-frank)
  - Merge blocked on #904 / #906 (@systemf-frank)
    - Merge blocked on #905 (@systemf-frank)
      - Merge blocked on #898 (@systemf-frank)
- #809 (@systemf-frank)
- #899 Has a quick fix, doesn't complain anymore about a missing main function, but results in a rather vague error message. I am not happy with this, so while the fix has been merged, I will leave this issue open so we won't forget. Can be moved to a future release if necessary. (@systemf-frank)

### Release Date

10th September.

**Comment by cwgoes** [comment]
Release cut - https://github.com/anoma/juvix/releases/tag/stygarctus.
